code/modules/mob/living/simple_animal/friendly/cat.dm

```diff
@@ -177,3 +177,92 @@
 	minbodytemp = 0
 	melee_damage_lower = 5
 	melee_damage_upper = 15
+
+/mob/living/simple_animal/pet/cat/cak
+	name = "Keeki"
+	desc = "It's a cat made out of cake. It looks like a brain would fit into it's head."
+	icon_state = "cak"
+	icon_living = "cak"
+	icon_resting = "cak_rest"
+	icon_dead = "cak_dead"
+	health = 50
+	maxHealth = 50
+	harm_intent_damage = 10
+	butcher_results = list(
+		/obj/item/organ/internal/heart = 1, 
+		/obj/item/reagent_containers/food/snacks/birthdaycakeslice = 3,  
+		/obj/item/reagent_containers/food/snacks/meat = 2
+	)
+	response_harm = "takes a bite out of"
+	attacked_sound = "sound/items/eatfood.ogg"
+	deathmessage = "loses its false life and collapses!"
+	death_sound = "bodyfall"
+	var/obj/item/organ/internal/brain/brain = null
+
+/mob/living/simple_animal/pet/cat/cak/Life()
+	..()
+	if(stat)
+		return
+	if(health < maxHealth)
+		adjustBruteLoss(-4)
+	for(var/obj/item/reagent_containers/food/snacks/donut/D in range(1, src))
+		if(D.icon_state != "donut2")
+			D.name = "frosted donut"
+			D.icon_state = "donut2"
+			D.reagents.add_reagent("sprinkles", 2)
+			D.filling_color = "#FF69B4"
+
+/mob/living/simple_animal/pet/cat/cak/attackby(obj/item/O, mob/living/user, params)
+	if(istype(O, /obj/item/organ/internal/brain))
+		var/obj/item/organ/internal/brain/B = O
+		if(brain) //There's already a brain in it.
+			to_chat(user, "<span class='warning'>There's already a brain in [src]!</span>")
+			return
+		if(!B.brainmob.key)
+			var/ghost_can_reenter = 0
+			if(B.brainmob.mind)
+				for(var/mob/dead/observer/G in GLOB.player_list)
+					if(G.can_reenter_corpse && G.mind == B.brainmob.mind)
+						ghost_can_reenter = 1
+						break
+				for(var/mob/living/simple_animal/S in GLOB.player_list)
+					if(S in GLOB.respawnable_list)
+						ghost_can_reenter = 1
+						break
+			if(!ghost_can_reenter)
+				to_chat(user, "<span class='notice'>[B] is completely unresponsive; there's no point.</span>")
+				return
+
+		to_chat(user, "<span class='notice'>You install [B] in [src]!</span>")

```

Comment:
I think install is a little weird for a food being. Can't just say "stick into" instead?